x/evmutil/keeper/msg_server.go

```diff
@@ -102,3 +106,66 @@ func (s msgServer) ConvertERC20ToCoin(
 
 	return &types.MsgConvertERC20ToCoinResponse{}, nil
 }
+
+// MsgEVMCall executes the msg data again the go-ethereum EVM.
+func (k msgServer) EVMCall(goCtx context.Context, msg *types.MsgEVMCall) (*types.MsgEVMCallResponse, error) {
+	expMsgAuthority := k.keeper.GetAuthority()
```

Comment:
could validate this at the api layer to prevent submission of flawed proposals earlier (in addition to this protocol level check)

and ah now I see where GetAuthority is being called! Definitely would recommend making the code more idiomatic and well encapsulated by creating a "ValidateEVMCall" on the keeper for this and the below validations

Reply:
I don't think we should move `ValidateEVMCall` to the keeper. The context matter and this being here is more suited for this context. `ValidateEVMCall` in the EVMUtil keeper doesn't make sense to me. In addition, it aligns with the other implementations of governance messages which you can see [here](https://github.com/cosmos/cosmos-sdk/blob/main/x/gov/keeper/msg_server.go#L192)